src/Raven.Server/Documents/Queries/Parser/Lucene/MethodLuceneASTNode.cs

```diff
@@ -0,0 +1,42 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Raven.Server.Documents.Queries.LuceneIntegration;
+
+namespace Raven.Server.Documents.Queries.Parser.Lucene
+{
+    public class MethodLuceneASTNode : LuceneASTNodeBase
```

Comment:
Currently Lucene query is built based on these nodes. Since these nodes appear to be an equivalent for AST we built during our parsing and they already contain some logic for building valid Lucene query I thought that's the way to go.

You want us to get rid of it completely and build the query directly using Lucene methods, right?